app/controllers/topics_controller.rb

```diff
@@ -1,2 +1,11 @@
 class TopicsController < ApplicationController
+  def index
+    @title = "Lesson Topics"
+    @topics = Topic.all
+  end
+
+  def show
+    @topic = Topic.friendly.find(params[:id])
+    @lesson = @topic.lessons.first || not_found
```

Comment:
Not sure whether you were deliberately avoiding this, but FYI in case not: `#take!` instead of `#first` here would raise `ActiveRecord::RecordNotFound` and create a 404.